Unable to access serial ports

<a alt="bounty" href="https://www.bountysource.com/issues/86212593"><img src="https://api.bountysource.com/badge/issue?issue_id=86212593"></a>

I am having no luck under docker

* Platform information:
  * Hardware: amd64 docker image
  * OS: Debian 10
  * Java Runtime Environment: Whatever docker snapshot has
  * openHAB version: 2.5.0
* Issue of the topic: 

Unable to detect nortek dongle for either communication protocol. "Offline - Communication Error"

The dropdown under the controller/coordinator does not show any serial dropdown like i see on videos. Im guessing something is missing. I have read over and over all the docs with no luck. Here is what i tried

 - Added openhab to groups on host: openhab tty dialout audio bluetooth
 - Confirmed serial device is using proper driver "cp210x" with "dmesg | grep tty"
    [    5.148772] usb 1-1.1.3: cp210x converter now attached to ttyUSB0
    [    5.160085] usb 1-1.1.3: cp210x converter now attached to ttyUSB1
 - Added EXTRA_JAVA_OPTS=-Dgnu.io.rxtx.SerialPorts=/dev/ttyUSB0:/dev/ttyUSB1
 - Docker device passed: --device=/dev/ttyUSB0 and --device=/dev/ttyUSB0
 - Alternatively tried passing devices as volume path as well seperately
 - Added zigbee as Port=/dev/ttyUSB1, flow control=software, baud=57600
 - Checked to see if docker can see the file using "docker exec -it openhab ls -lisa /dev/ttyUSB*
    11817 0 crw-rw-rw- 1 root dialout 188, 0 Dec 30 13:00 /dev/ttyUSB0
    16696 0 crw-rw-rw- 1 root dialout 188, 1 Dec 30 13:00 /dev/ttyUSB1
 - Checked docker uid and gid, both are 1000 (openhab)
 - Changed udev rules to 666 for ttyUSB[0-9]
 - Started home assistant docker image and zwave/zigbee both work and shows up with --device

Logs:

```
==> /srv/dev-disk-by-id-md-name-omv-Raid-part1/openhab/userdata/logs/events.log <==
2019-12-30 09:15:48.851 [hingStatusInfoChangedEvent] - 'zigbee:coordinator_ember:d407648b' changed from UNINITIALIZED to INITIALIZING
2019-12-30 09:15:48.858 [me.event.ThingUpdatedEvent] - Thing 'zigbee:coordinator_ember:d407648b' has been updated.
2019-12-30 09:15:48.860 [me.event.ThingUpdatedEvent] - Thing 'zigbee:coordinator_ember:d407648b' has been updated.
2019-12-30 09:15:48.861 [me.event.ThingUpdatedEvent] - Thing 'zigbee:coordinator_ember:d407648b' has been updated.
2019-12-30 09:15:48.864 [me.event.ThingUpdatedEvent] - Thing 'zigbee:coordinator_ember:d407648b' has been updated.
2019-12-30 09:15:48.865 [hingStatusInfoChangedEvent] - 'zigbee:coordinator_ember:d407648b' changed from INITIALIZING to UNKNOWN
2019-12-30 09:15:48.865 [nt.FirmwareStatusInfoEvent] - Firmware status of thing zigbee:coordinator_ember:d407648b changed to UNKNOWN.

==> /srv/dev-disk-by-id-md-name-omv-Raid-part1/openhab/userdata/logs/openhab.log <==
2019-12-30 09:15:49.864 [ERROR] [zigbee.dongle.ember.ZigBeeDongleEzsp] - Unable to open Ember serial port

==> /srv/dev-disk-by-id-md-name-omv-Raid-part1/openhab/userdata/logs/events.log <==
2019-12-30 09:15:49.865 [hingStatusInfoChangedEvent] - 'zigbee:coordinator_ember:d407648b' changed from UNKNOWN to OFFLINE: Failed to open communications port
2019-12-30 09:15:49.866 [hingStatusInfoChangedEvent] - 'zigbee:coordinator_ember:d407648b' changed from OFFLINE: Failed to open communications port to OFFLINE (COMMUNICATION_ERROR)

==> /srv/dev-disk-by-id-md-name-omv-Raid-part1/openhab/userdata/logs/openhab.log <==
2019-12-30 09:15:50.867 [ERROR] [zigbee.dongle.ember.ZigBeeDongleEzsp] - Unable to open Ember serial port

==> /srv/dev-disk-by-id-md-name-omv-Raid-part1/openhab/userdata/logs/events.log <==
2019-12-30 09:15:50.868 [hingStatusInfoChangedEvent] - 'zigbee:coordinator_ember:d407648b' changed from OFFLINE (COMMUNICATION_ERROR) to OFFLINE: Failed to open communications port
2019-12-30 09:15:50.869 [hingStatusInfoChangedEvent] - 'zigbee:coordinator_ember:d407648b' changed from OFFLINE: Failed to open communications port to OFFLINE (COMMUNICATION_ERROR)

==> /srv/dev-disk-by-id-md-name-omv-Raid-part1/openhab/userdata/logs/openhab.log <==
2019-12-30 09:15:55.869 [ERROR] [zigbee.dongle.ember.ZigBeeDongleEzsp] - Unable to open Ember serial port

==> /srv/dev-disk-by-id-md-name-omv-Raid-part1/openhab/userdata/logs/events.log <==
2019-12-30 09:15:55.870 [hingStatusInfoChangedEvent] - 'zigbee:coordinator_ember:d407648b' changed from OFFLINE (COMMUNICATION_ERROR) to OFFLINE: Failed to open communications port
2019-12-30 09:15:55.871 [hingStatusInfoChangedEvent] - 'zigbee:coordinator_ember:d407648b' changed from OFFLINE: Failed to open communications port to OFFLINE (COMMUNICATION_ERROR)
```

--- a/README.md
+++ b/README.md
@@ -26,10 +26,11 @@ Table of Contents
          * [User and group identifiers](#user-and-group-identifiers)
          * [Java cryptographic strength policy](#java-cryptographic-strength-policy)
       * [Parameters](#parameters)
-         * [Passing devices with symlinks](#Passing-devices-with-symlinks)
+         * [Passing devices with symlinks](#passing-devices-with-symlinks)
       * [Upgrading](#upgrading)
       * [Building the images](#building-the-images)
       * [Executing shell scripts before openHAB is started](#executing-shell-scripts-before-openhab-is-started)
+      * [Common problems](#common-problems)
       * [Contributing](#contributing)
       * [License](#license)
 
@@ -413,7 +414,7 @@ The following addons are known to depend on the unlimited cryptographic strength
 * `-v /openhab/userdata` - openHAB userdata directory
 * `--device=/dev/ttyUSB0` - attach your devices like RFXCOM or Z-Wave Sticks to the container
 
-#### Passing devices with symlinks
+### Passing devices with symlinks
 
 On Linux, if you pass a device with a symlink or any non standard name (e.g. /dev/ttyZWave), some addons require the device name to follow the Linux serial port naming rules (e.g. "ttyACM0", "ttyUSB0" or "ttyUSB-9999") or will otherwise fail to discover the device.
 
@@ -556,6 +557,42 @@ fi
 ```shell
 setcap 'cap_net_bind_service=+ep' "${JAVA_HOME}/bin/java"
 ```
+
+## Common problems
+
+### Error: KARAF_ETC is not valid
+
+This error message indicates that the data in the volumes is not properly initialized.
+The error is usually the result of a permissions issue or already putting files into volumes without the volumes having been initialized by the container first.
+When mounting directories into the container, check that these directories exist, are completely empty and are owned by the same `USER_ID` and `GROUP_ID` as configured in the container ENV variables.
+
+### Missing some preinstalled package
+
+Docker containers are kept as small as possible intentionally to decrease download times and the number of potential vulnerabilities for everyone.
+If you want additional packages installed, create your own container based on this container.
+Another option is to install the package by [executing a shell script before openHAB is started](#executing-shell-scripts-before-openhab-is-started).
+
+### No logging after "Launching the openHAB runtime..."
+
+By default this will always be the last logged message.
+A console logger can be [configured](#server-mode) for more detailed logging.
+
+### OpenMediaVault
+
+The default filesystem mount flags of OpenMediaVault contain the `noexec` flag which interferes with the serial library used by openHAB.
+To be able to use serial devices with openHAB, make sure the `userdata` volume mounted by the container is not backed by a filesystem having the `noexec` flag.
+See the [OMV documentation](https://openmediavault.readthedocs.io/en/5.x/various/fs_env_vars.html) on how to remove the `noexec` flag from an existing filesystem.
+
+### Portainer
+
+The default values of ENV variables are always stored by Portainer (see [portainer/portainer#2952](https://github.com/portainer/portainer/issues/2952)).
+This causes issues such as endless restart loops when upgrading the container with Portainer.
+To resolve this issue when upgrading openHAB, first remove all default (non-overriden) ENV variables before starting the new container.
+
+### SELinux
+
+When using the container on a Linux distribution with SELinux enabled (CentOS/Fedora/RHEL), add the `:z` or `:Z` option to volumes to give the container write permissions.
+For more information on this see the [Docker documentation](https://docs.docker.com/storage/bind-mounts/#configure-the-selinux-label).
 
 ## Contributing
 

